pkg/utils/node/finalizer.go

```diff
@@ -0,0 +1,17 @@
+package node
+
+import (
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+)
+
+func RemoveFinalizer(node *v1.Node, index int) {
+	if node.Finalizers == nil || index < 0 {
+		return
+	}
+	if node.Finalizers[index] == v1alpha1.KarpenterFinalizer {
+		node.Finalizers[index] = node.Finalizers[len(node.Finalizers)-1]
```

Comment:
This doesn't maintain the order of the finalizers. Is that important? (I'm not sure but wanted to ask)

Reply:
Nope. Order doesn't matter in finalizers since they are supposed to run asynchronously and be removed asynchronously be various controllers.